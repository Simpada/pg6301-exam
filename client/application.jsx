import {BrowserRouter, Link, Route, Routes} from "react-router-dom";
import {FrontPage} from "./pages/frontPage";
import {Menu} from "./pages/menu";
import {AddNewItem} from "./pages/addNewItem";
import React, {useContext} from "react";



export function Application() {


    return <BrowserRouter>
        <header>
            <Link to={"/"}>Front page</Link>
            <br/>
            <Link to={"/menu"}>Menu</Link>
            <div className="menu-divider" />
            {/*<UserActions user={data?.user} />*/}
        </header>
        <main>
            <Routes>
                <Route path={"/"} element={<FrontPage/>}/>
                <Route path={"/menu"} element={<Menu/>}/>
                <Route path={"/menu/add"} element={<AddNewItem/>}/>
                {/*<Route path={"/login/*"} element={<LoginPage config={data.config} reload={reload} />} />*/}
                {/*<Route path={"/profile"} element={<Profile user={data?.user} />} />*/}
                <Route path={"*"} element={<h1>Not found</h1>} />
            </Routes>
        </main>

    </BrowserRouter>

}