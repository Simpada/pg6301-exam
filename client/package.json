{
  "name": "client",
  "version": "1.0.0",
  "description": "",
  "scripts": {
    "build": "npm install --include-dev && parcel build index.html",
    "dev": "parcel watch index.html",
    "test": "jest"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "dependencies": {
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-router-dom": "^6.4.3"
  },
  "devDependencies": {
    "@babel/preset-env": "^7.20.2",
    "@babel/preset-react": "^7.18.6",
    "assert": "^2.0.0",
    "babel-jest": "^29.3.1",
    "console-browserify": "^1.2.0",
    "jest": "^29.3.1",
    "jest-environment-jsdom": "^29.3.1",
    "parcel": "^2.8.0",
    "process": "^0.11.10",
    "util": "^0.12.5"
  },
  "babel": {
    "presets": [
      "@babel/preset-react",
      [
        "@babel/preset-env",
        {
          "targets": {
            "node": "current"
          }
        }
      ]
    ]
  },
  "jest": {
    "testEnvironment": "jsdom",
    "collectCoverageFrom": [
      "**/*.jsx",
      "**/*.js",
      "!dist/**",
      "!coverage/**",
      "!lib/**"
    ]
  }
}
