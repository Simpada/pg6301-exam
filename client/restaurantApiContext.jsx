import React from "react";
import {fetchJSON} from "./lib/fetchJSON";
import {postJSON} from "./lib/postJSON";


export const RestaurantApiContext = React.createContext({
    async fetchLogin() {
        return await fetchJSON("/api/login");
    },
    async menu(query) {
        return await fetchJSON("/api/menu?" + new URLSearchParams(query));
    },
    async addItem(item) {
        return await postJSON("/api/menu/add", item);
    }
})