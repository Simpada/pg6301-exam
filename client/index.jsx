import * as React from "react";
import {createRoot} from "react-dom/client";
import {Application} from "./application";

const element = document.getElementById("app");
const root = createRoot(element);

root.render(<Application/>)