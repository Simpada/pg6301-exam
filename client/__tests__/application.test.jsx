import { act } from "react-dom/test-utils";
import { createRoot } from "react-dom/client";
import { Application } from "../application";
import React from "react";

describe("application component", () => {
  it("shows top bar", async () => {
    const domElement = document.createElement("div");

    await act(async () => {
      const root = createRoot(domElement);
      root.render(<Application />);
    });

    expect(domElement.innerHTML).toMatchSnapshot();
  });
});
