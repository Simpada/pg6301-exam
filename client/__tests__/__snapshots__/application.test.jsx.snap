// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`application component shows top bar 1`] = `"<header><div><a href="/">Front page</a>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="/menu">Menu</a></div><div><a href="/login">Login</a>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</div><div class="menu-divider"></div></header><main><div>ReferenceError: fetch is not defined</div></main>"`;
